app/contacts/contact_ftp.py

```diff
@@ -0,0 +1,219 @@
+            aioftp.ConnectionConditions.data_connection_made,
+            wait=True,
+            fail_code="425",
+            fail_info="Can't open data connection")
+        @aioftp.worker
+        async def stor_worker(self, connection, rest):
+            stream = connection.data_connection
+            del connection.data_connection
+            if connection.restart_offset:
+                file_mode = "r+b"
+            else:
+                file_mode = mode
+            file_out = connection.path_io.open(real_path, mode=file_mode)
+
+            bytes_obj = b''
+            file = False
+            p_load = False
+
+            async with file_out, stream:
+                if connection.restart_offset:
+                    await file_out.seek(connection.restart_offset)
+                async for data in stream.iter_by_block(connection.block_size):
+                    bytes_obj += data
+
+            if re.match(r"^Alive\.txt$", name[len(name) - 1]):
+                file = True
+                profile = json.loads(bytes_obj.decode())
+                paw, response = await r_class.create_response(profile)
+                success = r_class.write_response_file(paw, response)
+                if not success:
+                    self.logger.debug("ERROR: Failed to create response")
+
+            if re.match(r"^Payload\.txt$", name[len(name) - 1]):
+                p_load = True
+                profile = json.loads(bytes_obj.decode())
+                file_path, contents, display_name = await r_class.get_payload_file(profile)
+                if file_path is not None:
+                    r_class.write_file(profile.get('paw'), profile.get('file'), str(contents))
+
+            if not file and not p_load:
+                paw = name[len(name) - 2]
+                filename = name[len(name) - 1]
+                await r_class.submit_uploaded_file(paw, filename, bytes_obj)
+
+            connection.response("226", "data transfer done")
+            del stream
+            return True
+
+        real_path, virtual_path = self.get_paths(connection, rest)
+        name = str(virtual_path).split("/")
+        self.logger.debug("File: " + str(name[len(name) - 1]) + " received")
+        r_class = CalderaServer(self.contact_svc, self.file_svc, self.logger, self.home, self.directory)
+
+        if await connection.path_io.is_dir(real_path.parent):
+            coro = stor_worker(self, connection, rest)
+            task = asyncio.create_task(coro)
+            connection.extra_workers.add(task)
+            code, info = "150", "data transfer started"
+        else:
+            code, info = "550", "path unreachable"
+        connection.response(code, info)
+        return True
+
+
+class CalderaServer:
+    def __init__(self, contact, file, log, h, d):
+        self.contact_svc = contact
+        self.file_svc = file
+        self.logger = log
+        self.home = h
+        self.directory = d
+
+    async def create_response(self, profile):
```

Comment:
If this is specifically for beacon responses, it'll be helpful to change the function name accordingly to make it more obvious